src/main/main.gd

```diff
@@ -7,14 +7,32 @@ extends Node2D
 var in_cutscene := false
 
 var maps:Array = [preload('res://src/maps/map-1.tscn'),preload('res://src/maps/map-2.tscn'),preload('res://src/maps/map-3.tscn')]
+var map_room_id_dict: Dictionary = { #Not perfect but works for time constraint
```

Comment:
This dictionary is redundant as we could use the index of the maps array above it. 

Although considering all the maps could follow the "map-x.tscn" name convention we wouldn't need to load them all into anything, as we could just preload 'res://src/maps/map-{room_id}.tscn'.

if you want to make that change now great, otherwise we'll do it soon